javaslang/src/main/java/javaslang/collection/CharSeq.java

```diff
@@ -22,19 +24,32 @@
  * The CharSeq (read: character sequence) collection essentially is a rich String wrapper having all operations
  * we know from the functional Javaslang collections.
  *
- * @author Ruslan Sennov, Daniel Dietrich
+ * @author Ruslan Sennov, Daniel Dietrich, Pap Lőrinc
  * @since 2.0.0
  */
 public final class CharSeq implements Kind1<CharSeq, Character>, CharSequence, IndexedSeq<Character>, Serializable {
-
     private static final long serialVersionUID = 1L;
 
-    private static final CharSeq EMPTY = new CharSeq("");
+    private static final CharSeq EMPTY = new CharSeq(Vector.empty());
+
+    private final Vector<Character> delegate;
```

Comment:
// TODO investigate the effect of storing bytes in case of ASCII chars
http://openjdk.java.net/jeps/254